### ISSUE TITLE
Hyperparameter optimization on SLURM workload manager

### ISSUE THREAD

**Issue by YunchaoYang** [YunchaoYang/Blogs#42]
Hyperparameter optimization (HPO) is for tuning the hyperparameters of your machine learning model, e.g., the learning rate, filter sizes, etc. There are several popular algorithms used for HPO including grid search, random search, Bayesian optimization, and genetic optimization. Similarly, there are several libraries and tools implementing these algorithms, each having their own tradeoffs in usability, flexibility, and feature support.

A hyperparameter optimization framework can greatly alleviate this common pain point, as it can:

"Automatically run a small-scale experiments quickly to identify key performance-influencing hyperparameters.
Use smart optimization algorithms to search for optimal hyperparameter sets, which is more efficient than random search or grid search, let alone manual search."

- Resources 

  - [NERSC slurm raytune](https://docs.nersc.gov/machinelearning/hpo/#raytune) https://github.com/NERSC/slurm-ray-cluster

  - [ALCF DeepHyper](https://deephyper.github.io/events/workshop-anl-2022-summer)
